File: benchmarks/solutions/swift/a3-valid-parens.swift
func isValid(_ s: String) -> Bool {
    var stack: [Character] = []
    for c in s {
        switch c {
        case "(", "[", "{": stack.append(c)
        case ")": guard stack.popLast() == "(" else { return false }
        case "]": guard stack.popLast() == "[" else { return false }
        case "}": guard stack.popLast() == "{" else { return false }
        default: break
        }
    }
    return stack.isEmpty
}

File: benchmarks/solutions/kotlin/a3-valid-parens.kt
fun isValid(s: String): Boolean {
    val stack = ArrayDeque<Char>()
    val pairs = mapOf(')' to '(', ']' to '[', '}' to '{')
    for (c in s) {
        if (c in pairs) {
            if (stack.removeLastOrNull() != pairs[c]) return false
        } else {
            stack.addLast(c)
        }
    }
    return stack.isEmpty()
}
